web/src/views/FileBrowserView/FileBrowser.vue

```diff
@@ -377,18 +377,38 @@ const isOwner = computed(() => !!(
 ));
 const itemsNotFound = computed(() => items.value && !items.value.length);
 
+function replaceStringsInEndpoint(
+  endpoint: string,
+  substitutions: {[key: string]: string} | undefined,
+) {
+  if (!substitutions) return endpoint;
+  let result = endpoint;
+  Object.entries(substitutions).forEach(([key, value]) => {
+    result = result.replace(key, value);
+  });
+  return result;
+}
+
 function getExternalServices(path: AssetPath) {
   const servicePredicate = (service: Service, _path: AssetPath) => (
     new RegExp(service.regex).test(path.path)
           && _path.asset !== null
           && _path.aggregate_size <= service.maxsize
   );
 
+  const baseApiUrl = trimEnd(process.env.VUE_APP_DANDI_API_ROOT, '/');
+
+  const substitutions = path.asset ? {
+    $asset_s3_download_url$: trimEnd(path.asset.url, '/'),
+    $asset_api_download_url$: `${baseApiUrl}/assets/${path.asset.asset_id}/download/`,
```

Comment:
@waxlamp - actually one of the things i wanted in the other PR is the dandiset specific asset download url, not the general one. that would allow in essence for someone to fetch relevant dandiset info or for externallinks such as videos, the video url. also since this is clicked not from a global search but a view within the dandiset, i think it's appropriate to have the other url.

Reply:
> the other PR

Do you mean #1655 or #1692?

> dandiset specific asset download url, not the general one

When you say "dandiset specific", do you mean the longer URL that is "nested" under a `/dandisets/{dandiset_id}/...` URL?

Reply:
correct - sorry was in meetings. yes, i would like that asset url to change to the one that is nested under /dandisets/